Mage/src/main/java/mage/game/events/RemoveCounterEvent.java

```diff
@@ -0,0 +1,33 @@
+package mage.game.events;
+
+import mage.abilities.Ability;
+import mage.cards.Card;
+import mage.players.Player;
+
+public class RemoveCounterEvent extends GameEvent {
+
+    boolean isDamage;
+
+    public RemoveCounterEvent(String name, Card targetCard, Ability source, boolean isDamage){
+        super(GameEvent.EventType.REMOVE_COUNTER, targetCard.getId(), source,
+                targetCard.getControllerOrOwnerId());
+
+        if (source != null && source.getControllerId() != null) {
+            setPlayerId(source.getControllerId()); // player who controls the source ability that removed the counters
```

Comment:
Same with `getControllerOrOwnerId` -- see above

Reply:
looks like this didn't get addressed?

Reply:
Yes, it’s not fixed yet — must save only one logical value to playerId, not two different (can be: null + controller, but not: owner + controller):

![IMG_0285](https://github.com/magefree/mage/assets/8344157/66aae40d-9532-4510-8a03-7749834fa628)